### PR TITLE
feat(web): sort and column function in views

### PR DESCRIPTION
**Pull request by coolguy741**
# Overview
added sorting and column function in views

### CODE REVIEW

File: server/internal/infrastructure/mongo/item.go [hunk 1]
```diff
@@ -406,6 +406,113 @@ func fieldKey(f view.FieldSelector) string {
...
+	return ff
+}
+
+func fieldKey(f view.FieldSelector) string {
+	if f.Type == view.FieldTypeMetaField || f.Type == view.FieldTypeField {
+		return "__temp." + f.ID.String()
+	}
+	return string(f.Type)
+}
+
```
Comment (nourbalaha) [inline server/internal/infrastructure/mongo/item.go:515]: Please revert the server changes as this is a web PR and it is duplicated.

File: web/src/components/molecules/Content/Table/index.tsx [hunk 2]
```diff
@@ -92,6 +96,8 @@ const ContentTable: React.FC<Props> = ({
   pageSize,
   requests,
   addItemToRequestModalShown,
+  columns,
+  setColumns,
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/Table/index.tsx:100]: Instead of passing these as props you can move the useState from the hooks to this file since it is only used here.

File: web/src/gql/queries/item.ts [hunk 3]
```diff
@@ -121,43 +121,45 @@ export const GET_ITEMS_BY_IDS = gql`
...
           url
         }
-        createdBy {
-          ... on Integration {
-            name
-          }
-          ... on User {
-            name
-          }
-        }
```
Comment (nourbalaha) [inline web/src/gql/queries/item.ts:149]: Why did you remove createdBy?

File: web/src/gql/queries/item.ts [hunk 4]
```diff
@@ -206,7 +208,7 @@ export const UPDATE_ITEM = gql`
         schemaId
         fields {
           value
-          type
+          types
```
Comment (nourbalaha) [inline web/src/gql/queries/item.ts:211]:
```suggestion
          type
```

File: web/src/components/molecules/Content/Table/index.tsx [hunk 5]
```diff
@@ -509,15 +566,28 @@ const ContentTable: React.FC<Props> = ({
           dataSource={contentTableFields}
           tableAlertOptionRender={AlertOptions}
           rowSelection={rowSelection}
-          columns={[...actionsColumn, ...contentTableColumns]}
+          columns={[...actionsColumn, ...contentColumn]}
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/Table/index.tsx:569]: This line is giving an error because the contentColumn might be undefined.
add something like this:
```
const tableColumns = useMemo(() => {
    const items = [...actionsColumn];
    if (contentColumn) items.push(...contentColumn);
    return items;
  }, [actionsColumn, contentColumn]);
  ```

File: web/src/components/molecules/Content/Table/index.tsx [hunk 6]
```diff
@@ -509,15 +566,28 @@ const ContentTable: React.FC<Props> = ({
...
+          columnsState={{
+            value: columns,
+            onChange: setColumns,
+          }}
           onChange={(pagination, _, sorter: any) => {
+            console.log("sorter", sorter);
+            console.log("contentTableFields", contentTableFields);
+            console.log("contentTableColumns", contentTableColumns);
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/Table/index.tsx:577]: remove `console.log` statements from PRs.

File: web/src/components/molecules/Content/Table/index.tsx [hunk 7]
```diff
@@ -196,6 +238,21 @@ const ContentTable: React.FC<Props> = ({
     [t, sort?.field?.type, sort?.direction, selectedItem?.id, onItemSelect],
   );
 
+  const contentColumn: ProColumns<ContentTableField>[] | undefined = useMemo(
```
Comment (nourbalaha) [inline web/src/components/molecules/Content/Table/index.tsx:241]:
```suggestion
  const contentColumns: ProColumns<ContentTableField>[] | undefined = useMemo(
```

**Review by nourbalaha — CHANGES_REQUESTED** [review]

File: web/src/components/molecules/Content/Table/index.tsx [hunk 8]
```diff
@@ -196,6 +238,21 @@ const ContentTable: React.FC<Props> = ({
...
+    () =>
+      contentTableColumns?.map(column => ({
```
Comment (caichi-t) [inline web/src/components/molecules/Content/Table/index.tsx:243]: I think it's better that adding property to contentTableColumns in src\components\organisms\Project\Content\ContentList\hooks.ts, but if you did that, is there any problems?

File: web/src/components/molecules/Content/Table/index.tsx (hunk 2 above)
Comment (coolguy741) [reply web/src/components/molecules/Content/Table/index.tsx:100]: It's because it will be used for integrating into views menu

File: web/src/gql/queries/item.ts (hunk 3 above)
Comment (coolguy741) [reply web/src/gql/queries/item.ts:149]: When graphQL API that requests createdBy user quits the local server(http://localhost:8080/) on my local environment. 
It will add createBy and updatedBy when this is fixed.

File: web/src/gql/queries/item.ts (hunk 4 above)
Comment (coolguy741) [reply web/src/gql/queries/item.ts:211]: Is it changed to "types"?

File: web/src/components/molecules/Content/Table/index.tsx (hunk 8 above)
Comment (coolguy741) [reply web/src/components/molecules/Content/Table/index.tsx:243]: No problem on that, I changed it because it would affect the filtering.
If filtering is ok, we can use "ProColumns<ContentTableField>[] | undefined".

Comment (caichi-t) [reply web/src/components/molecules/Content/Table/index.tsx:243]: Filtering is ok! You can add the properties in hooks.ts!

File: web/src/components/molecules/Content/Table/index.tsx (hunk 2 above)
Comment (nourbalaha) [reply web/src/components/molecules/Content/Table/index.tsx:100]: OK, so you can keep it the way it is.

File: web/src/components/molecules/Content/Table/index.tsx (hunk 8 above)
Comment (coolguy741) [reply web/src/components/molecules/Content/Table/index.tsx:244]: I did this to separate data fields and visualization fields.

**Review by nourbalaha — APPROVED** [review]
